### DISCUSSION TITLE
What is the minimum requirement to create a new mixin from the skeletor template

### DISCUSSION THREAD

**Discussion by richbai90** [getporter/porter#2170] [Q&A]
I followed the readme and made all the changes mentioned. The mixin builds and installs fine, but building a bundle with the mixin fails with the error `* mixin execution failed: package not found (/cnab/app/mixins/xfer/runtimes/xfer-runtime).` What is required for this runtime to be generated?

**Comment by carolynvs** [comment]
What you did is sufficient. Can you run the following command and check that the mixin is installed properly?

```
stat ~/.porter/mixins/xfer
stat ~/.porter/mixins/xfer/runtimes/xfer-runtime
```

When you include a mixin in a bundle (below), Porter automatically copies the runtime binary into the bundle in the path that is failing for you. I'm wondering if perhaps the binary wasn't installed properly when you ran `mage install`.

**Reply by richbai90** [reply]
The xfer binary was there, but the `/xfer/runtimes/xfer-runtime` binary wasn't. It was however in the build directory under `/bin/mixins/xfer/runtimes.` Copying that file to the `~/.porter/mxins/xfer/runtimes` folder fixed the issue. It may be a bug with porter V1. I'm running a fork of V1 currently that has the changes I made in #2167 .

**Reply by carolynvs** [reply]
The problem is most likely in the magefile.go in the `Install` function where we copy the binaries from the mixin's repository into your Porter installation directory.

The steps should be:

1. `mage build` compiles both the client and runtime binaries for your mixin
2. `mage install` copies both binaries into your PORTER_HOME directory (~/.porter)

If you are using the most recent code from skeletor, this is the implementation of the install target:

https://github.com/getporter/magefiles/blob/abc768862549990e7c7d96adc06aa81315fd02b7/mixins/magefile.go#L93-L100

All of the other mixins use this same target so I'm not quite sure why it's not working for you. If you can share a link to the code, I can take a further look.

**Reply by richbai90** [reply]
Sorry for not getting back. I noticed a few problems with my code which took a minute to resolve. Having resolved them, the issue remains. Here is the code: https://github.com/richbai90/porter-xfer-mixin

**Reply by carolynvs** [reply]
Oh my gosh, this is my fault. The skeletor repository is out of date. I'm real sorry about that, I just submitted a PR to your repo with the required changes to get it working.

https://github.com/richbai90/porter-xfer-mixin/pull/1

I'll get skeletor updated so that more people don't run into the same problem, thank you for pushing back on me insisting that it was working! 🤦‍♀️

**Reply by richbai90** [reply]
We've all been there. Thank you for the assistance. 😄
